BLUESPAWN-client/src/hunt/hunts/HuntT1035.cpp

```diff
@@ -46,35 +46,36 @@ namespace Hunts {
 
 			detections += 2;
 		}
-
+		
 		auto name = key.GetName();
 		name = name.substr(name.find_last_of(L"\\") + 1);
-
 		RegistryKey subkey = RegistryKey{ HKEY_LOCAL_MACHINE, L"SYSTEM\\CurrentControlSet\\Services\\" + name + L"\\Parameters" };
 
-		if (subkey.Exists() && subkey.ValueExists(L"ServiceDll")) {
-			auto filepath2 = GetImagePathFromCommand(subkey.GetValue<std::wstring>(L"ServiceDll").value());
-
-			if (!FileSystem::CheckFileExists(filepath2)) {
-				return detections;
+		for (auto regkey : { key, subkey }) {
+			if (!regkey.Exists()) {
+				continue;
 			}
+			for (auto value : { L"ServiceDll", L"ServiceMain", L"ServiceDllUnloadOnStop" }) {

```

Comment:
ServiceMain is an export inside ServiceDll. This should not be treated as a separate file. That said, there may be some sort of clever attack where a legitimate ServiceDll has an export that does something like disable the firewall or remove AV definitions.
ServiceDllUnloadOnStop specifies whether the ServiceDll should be unloaded from the instance of svchost when the service stops and is also not a file.

Reply:
Example of attack leveraging possible technique described above

ServiceDll = FirewallAPI.dll
ServiceMain = FWDeleteAllFirewallRules

Reply:
fixed